packages/jaeger-ui/src/components/SearchTracePage/SearchForm.jsx

```diff
@@ -259,44 +259,79 @@ export function submitForm(fields, searchTraces) {
 }
 
 export class SearchFormImpl extends React.PureComponent {
+  constructor(props) {
+    super(props);
+    this.state = {
+      formData: {
+        service: this.props.initialValues?.service,
+        operation: this.props.initialValues?.operation,
+        tags: this.props.initialValues?.tags,
+        lookback: this.props.initialValues?.lookback,
+        startDate: this.props.initialValues?.startDate,
+        startDateTime: this.props.initialValues?.startDateTime,
+        endDate: this.props.initialValues?.endDate,
+        endDateTime: this.props.initialValues?.endDateTime,
+        minDuration: this.props.initialValues?.minDuration,
+        maxDuration: this.props.initialValues?.maxDuration,
+        resultsLimit: this.props.initialValues?.resultsLimit,
+      },
+    };
+  }
+
+  handleChange = fieldData => {
+    this.setState(prevState => ({
+      formData: {
+        ...prevState.formData,
+        ...fieldData,
+      },
+    }));
+    if (fieldData.service) {
+      this.props.changeServiceHandler(fieldData.service);
+      this.setState(prevState => ({
+        formData: {
+          ...prevState.formData,
+          operation: DEFAULT_OPERATION,
+        },
+      }));
+    }
+  };
+
+  handleSubmit = e => {
+    e.preventDefault();
+    this.props.submitFormHandler(this.state.formData);
+  };
+
   render() {
-    const {
-      handleSubmit,
-      invalid,
-      searchMaxLookback,
-      selectedLookback,
-      selectedService = '-',
-      services,
-      submitting: disabled,
-    } = this.props;
+    const { invalid, searchMaxLookback, services, submitting: disabled } = this.props;
+    const { formData } = this.state;
+    const { service: selectedService, lookback: selectedLookback } = formData;
     const selectedServicePayload = services.find(s => s.name === selectedService);
     const opsForSvc = (selectedServicePayload && selectedServicePayload.operations) || [];
     const noSelectedService = selectedService === '-' || !selectedService;
     const tz = selectedLookback === 'custom' ? new Date().toTimeString().replace(/^.*?GMT/, 'UTC') : null;
 
     return (
-      <Form layout="vertical" onSubmitCapture={handleSubmit}>
+      <Form layout="vertical" onSubmitCapture={this.handleSubmit}>
         <FormItem
           label={
             <span>
               Service <span className="SearchForm--labelCount">({services.length})</span>
             </span>
           }
         >
-          <Field
+          <SearchableSelect
             name="service"
-            component={AdaptedSelect}
+            value={this.state.formData.service}
             placeholder="Select A Service"
-            props={{
-              disabled,
-            }}
+            disabled={disabled}
```

Comment:
where is `disabled` value defined? I see it being renamed as `submitting` - personally I think that should be how it is always called, because `submitting` reflects the state of the form, `disabled` is just a property of individual fields.